src/Server.php

```diff
@@ -10,25 +10,35 @@ class Server extends EventEmitter implements ServerInterface
 {
     public $master;
     private $loop;
+    private $isSecure = false;
 
     public function __construct(LoopInterface $loop)
     {
         $this->loop = $loop;
     }
 
-    public function listen($port, $host = '127.0.0.1')
+    public function listen($port, $host = '127.0.0.1', $streamContext = array())
     {
         if (strpos($host, ':') !== false) {
             // enclose IPv6 addresses in square brackets before appending port
             $host = '[' . $host . ']';
         }
+        if (isset($streamContext['ssl'])) {
+            $this->isSecure = true;
+        }
 
-        $this->master = @stream_socket_server("tcp://$host:$port", $errno, $errstr);
+        $this->master = @stream_socket_server(
+            "tcp://$host:$port",
+            $errno,
+            $errstr,
+            STREAM_SERVER_BIND | STREAM_SERVER_LISTEN,
+            stream_context_create($streamContext)
+        );
         if (false === $this->master) {
             $message = "Could not bind to tcp://$host:$port: $errstr";
             throw new ConnectionException($message, $errno);
         }
-        stream_set_blocking($this->master, 0);
+        stream_set_blocking($this->master, (int) $this->isSecure);
```

Comment:
There is a way to do it... I don't have react specific code samples for it, however, the gist of it is to try until `stream_socket_enable_crypto` returns success.

```
		$result = stream_socket_enable_crypto($this->stream, true, $crypto_method);
		if ($result === 0)
			return true;

		if ($result === false)
			return false;

		$this->secure = self::STATE_ENCRYPTED;
		return true;
```

Assume the above to be part of a member function which you keep polling until the value of secure == STATE_ENCRYPTED. You don't need to do blocking at all in this way, see documentation of `stream_socket_enable_crypto`. Depending on the `stream_select` timeout values typically doing it this way results in 1-3 cycles of polling to enable cryptography (tested on loopback interface).